alibi_detect/saving/loading.py

```diff
@@ -28,60 +28,38 @@
 
 logger = logging.getLogger(__name__)
 
-REQUIRES_BACKEND = [
-    'ClassifierDrift',
-    'ClassifierUncertaintyDrift',
-    'LearnedKernelDrift',
-    'LSDDDrift',
-    'MMDDrift',
-    'RegressorUncertaintyDrift',
-    'SpotTheDiffDrift'
-]
 
 # Fields to resolve in resolve_config ("resolve" meaning either load local artefact or resolve @registry, conversion to
 # tuple, np.ndarray and np.dtype are dealt with separately).
+# Note: For fields consisting of nested dicts, they must be listed in order from deepest to shallowest, so that the
+# deepest fields are resolved first. e.g. 'preprocess_fn.src' must be resolved before 'preprocess_fn'.
 FIELDS_TO_RESOLVE = [
-    ['preprocess_fn'],
     ['preprocess_fn', 'src'],
     ['preprocess_fn', 'model'],
     ['preprocess_fn', 'embedding'],
     ['preprocess_fn', 'tokenizer'],
     ['preprocess_fn', 'preprocess_batch_fn'],
+    ['preprocess_fn'],
     ['x_ref'],
+    ['c_ref'],
     ['model'],
     ['optimizer'],
     ['reg_loss_fn'],
-    ['kernel'],
     ['dataset'],
     ['kernel', 'src'],
     ['kernel', 'proj'],
+    ['kernel', 'init_sigma_fn'],
     ['kernel', 'kernel_a', 'src'],
     ['kernel', 'kernel_b', 'src'],
-    ['initial_diffs']
-]
-
-# Directories to amend before resolving config (fields to prepend config file dir to).
-# e.g. in a `config.toml` stored in `filepath='my_detector'` we might have `preprocess_fn.model.src = 'model/'.
-# Before resolution, we need to update `preprocess_fn.model.src` to be relative to runtime dir i.e. 'my_detector/model/'
-# Note some fields in FIELDS_TO_RESOLVE are missing from DIR_FIELDS, since some fields cannot be spec'd as directories.
-# e.g. embedding cannot be spec'd directly as a .dill file reference, it must be an EmbeddingConfig.
-DIR_FIELDS = [
```

Comment:
Can you summarize the new logic and how it's different/superior to the old one having `DIR_FIELDS` separately?

Reply:
Previously we did the following in `resolve_config`:

```python
        for key in DIR_FIELDS:
            src = _get_nested_value(cfg, key)
            if isinstance(src, str):
                src = config_dir.joinpath(Path(src))
                if src.is_file() or src.is_dir():
                    _set_nested_value(cfg, key, str(src))
        _prepend_cfg_filepaths(cfg, config_dir)
```

So we looped through `DIR_FIELDS` and prepended the `config.toml` directory to all the filepaths within it (so that we can do `x_ref = 'x_ref.npy'` in the `config.toml`, instead of having to do `x_ref = 'filepath/x_ref.npy'` where `filepath` is the path to the `config.toml` itself. 

`_prepend_cfg_filepaths` has now been modified so that it recurses through all fields in the config dict, attempts to prepend the `config.toml` filepath to any strings, and if the result of that gives a valid directory or file, it updates the config. In other words, we no longer have to specify `DIR_FIELDS` separatetly (the difference between this and `FIELDS_TO_RESOLVE` had caused some confusion previously, so not ideal to maintain both).